src/hotspot/share/opto/domgraph.cpp

```diff
@@ -233,9 +233,20 @@ uint Block_Stack::most_frequent_successor( Block *b ) {
   case Op_Jump:
   case Op_Root:
   case Op_Goto:
-  case Op_NeverBranch:
     freq_idx = 0;               // fall thru
     break;
+  case Op_NeverBranch: {
+    Node* succ = n->as_NeverBranch()->proj_out(0)->unique_ctrl_out();
+    int succ_idx = 0; // normal case
+    if (succ == b->_succs[1]->head()) {
+      // edges swapped, rare case
+      succ_idx = 1;
+    } else {
```

Comment:
Could you explain how do we end up at this rare case? In my opinion, it would be preferable, if possible, to ensure edges are never swapped in the first place.

Reply:
Never mind, I found the answer in https://github.com/openjdk/jdk/pull/11481. Maybe it would be worth extending the comment with a summary of why this can happen.

```suggestion
    if (succ == b->_succs[1]->head()) {
      // Edges swapped, rare case. May happen due to an unusual matcher
      // traversal order for peeled infinite loops.
      succ_idx = 1;
    } else {
```